aptos-move/aptos-vm/src/transaction_metadata.rs

```diff
@@ -55,6 +55,9 @@ impl TransactionMetadata {
                 .collect(),
             sequence_number: txn.sequence_number(),
             fee_payer: txn.authenticator_ref().fee_payer_address(),
+            // `fee_payer_authentication_key` is `None` if the TransactionAuthenticator
+            // lacks an authenticator for the fee payer. It is `Some([])` if the
+            // authenticator for the fee payer is a NoAccountAuthenticator.
             fee_payer_authentication_key: txn.authenticator().fee_payer_signer().map(|signer| {
```

Comment:
Can we have this a doc string `/// ... ` for the field in `TransactionMetadata`, rather then somewhere in the constructor?. Also let's add a unit test that checks this invariant

Reply:
Also, you can use `[TransactionAuthenticator]` to make structs searchable in IDE + when refactoring and checking usages (works with `///`).